packages/mui-material/src/Grid/Grid.js

```diff
@@ -193,7 +193,16 @@ const GridRoot = styled('div', {
       container && styles.container,
       item && styles.item,
       zeroMinWidth && styles.zeroMinWidth,
-      container && spacing !== 0 && styles[`spacing-xs-${String(spacing)}`],
+      container &&
+        spacing &&
+        spacing !== 0 &&
```

Comment:
```suggestion
```
This is redundant with the line before

Reply:
I don’t believe it’s redundant, if a variable exists but is undefined, the following condition evaluates to true:
 `myvariable !== 0`
 
Having only `spacing !== 0` as the condition was what was causing the class to parse as MuiGrud-spacing-undefined when spacing wasn’t specified. 

Should zero values for spacing be parsed into a class? I was under the impression from the original code that they should not be. (This made sense to me since the default spacing is zero, so, no need to specify it.)

That said, maybe the condition should be changed to 
`Number(spacing) !== 0`
to cover a user entering `spacing=“0”` (ie zero spacing but written as a string.)

Reply:
Sorry, am on my phone & misread which part of the condition was being suggested to be cut. 

Yes, having just `spacing` as the condition will work for all cases except `spacing=“0”` - and my code as written wouldn’t handle that case correctly anyway. 

I think my revised version would, but you’d probably know better whether that’s something that should be addressed or not.

Reply:
@Janpot I added an additional commit that addresses this.